### DISCUSSION TITLE
ReturnModelType results in any

### DISCUSSION THREAD

**Discussion by dor272** [typegoose/typegoose#452] [Q&A]
Hi, 
I've created a class and try to expose it's model type.
From the documentation I understood I'm supposed to use `ReturnModelType<typeof Class>`, but it results with the type `any`.

I also tried to use the[ example from the docs](https://typegoose.github.io/typegoose/docs/api/types/return-model-type/),
and it's the same, this type is `any`.

My class:
```ts
class Key {
   @prop()
    value?: string

   static getRandom(this: ReturnModelType<typeof Key>){
       return this.findOne({})
   }
}
```
versions:
mongoose - `5.11.7`
@types/mongoose - `5.10.3`
typegoose - `7.4.5`

Why does it happen? Am I missing something?

Thanks.

**Comment by hasezoey** [comment]
please provide your class (and where you get the type), and the versions you are using from: `mongoose @types/mongoose typegoose`

**Reply by dor272** [reply]
updated my original post

**Reply by hasezoey** [reply]
try downgrading your mongoose version to the supported range from typegoose (max mongoose version is `5.10.18`)

**Reply by dor272** [reply]
now it get recognized but I have another problem.
I use a mongoose plugin that have can have options passed to in queries function (like `find, findOne, etc...`).

before I extended mongoose interface like that:
```
declare module 'mongoose' {
    interface QueryOptions {
        authLevel?: any
    }
}
```

now I'm not sure what I'm suppose to extend

**Reply by hasezoey** [reply]
i cannot help with that, but currently typegoose only support mongoose 5.10.18, because all up have its own types - which are not fully compatible with typegoose's types
